docs/application/native/guides/applications/component-based-app.md

```diff
@@ -0,0 +1,642 @@
+    }
+
+    component_class_h __app_create_cb(void *user_data)
+    {
+         component_class_h comp_class = NULL;
+         comp_class = frame_component_add(comp_class, "base-frame", NULL);
+         comp_class = service_component_add(comp_class, "base-service", NULL);
+        return comp_class;
+    }
+
+    void __app_terminate_cb(void *user_data)
+    {
+    }
+
+    int main(int argc, char** argv)
+    {
+        int ret;
+        component_based_app_lifecycle_callback_s callback = {
+            .create = __app_create_cb,
+            .terminate = __app_terminate_cb
+        };
+
+        ret = component_based_app_main(argc, argv, &callback, NULL);
+        if (ret != APP_ERROR_NONE)
+            return ret;
+
+        return 0;
+    }
+    ```
+
+<a name="register_action"></a>
+## Managing Actions
+
+The Component-Based Application can register actions and receive action event. You can send an action event to another application using app-control.
+
+1. The app-control declaration in the manifest file
+    ```
+    <component-based-application appid="org.tizen.base-component" exec="@BINDIR@/base-component" nodisplay="false" multiple="false" type="capp">
+        <label>Base-component Application</label>
+        <icon>@DESKTOP_ICON@</icon>
+        <app-control id="dial">
+            <operation name="http://tizen.org/appcontrol/operation/dial"/>
+            <mime name="*"/>
+        </app-control>
+        <app-control id="dial-for-excel">
+            <operation name="http://tizen.org/appcontrol/operation/dial"/>
+            <mime name="application/vnd.ms-excel"/>
+        </app-control>
+    ```
+
+2.  Registering an action
+
+    To receive action events, each component instance should register the action.
+    You can registering an action as follows.
+    ```
+    static void __frame_component_action_cb(component_h context,
+            const char *action, app_control_h app_control,
+            void *user_data)
+    {
+    }
+
+    static Evas_Object *__frame_component_create_cb(component_h context, void *user_data)
+    {
+        component_register_action(context, "dial-for-excel")
+        return frame_get_window(ad->frame);
+    }
+
+    component_class_h app_control_component_add(component_class_h comp_class,
+            const char *component_id, void *user_data)
+    {
+        frame_component_lifecycle_callback_s callback = {
+            .create = __frame_component_create_cb,
+            .action = __frame_component_action_cb,
+        };
+
+        return component_based_app_add_frame_component(comp_class,
+                component_id, &callback, user_data);
+    }
+    ```
+
+3.  Sending an action event to another application
+
+    Actions are declared in the manifest files. To send a proper action, you should carefully set app-control values according to the manifest file app-control specification. In this example, the app-control will be set for "dial-for-excel" action.
```

Comment:
Please change "you should carefully set" to "you must set"

In this example, the app-control will be set for the **dial-for-excel** action.

Reply:
Done